compiler/rustc_parse/src/parser/stmt.rs

```diff
@@ -213,17 +209,22 @@ impl<'a> Parser<'a> {
         }
     }
 
+    fn emit_invalid_variable_declaration(
+        &self,
+        span: Span,
+        subdiagnostic: fn(Span) -> InvalidVariableDeclarationSub,
+    ) {
+        self.sess.emit_err(InvalidVariableDeclaration { span, sub: subdiagnostic(span) });
```

Comment:
Is this the correct way to emit a `SessionDiagnostic`? The blog post doesn't actually tell you what to do with the type once you've created it.

Reply:
It is, yeah. Might be worth updating the post given that's quite the omission, but hopefully the linked examples help.

Given that this function is just a one-liner, you could just remove it and inline this at the call sites.

Reply:
Makes sense, I've inlined it.